### PR TITLE
feat: Add SDK id and version to requests

### PR DESCRIPTION
**Pull request by fabriziodemaria**
To comply with the updated Confidence APIs.

### CODE REVIEW

File: Provider/src/main/java/com/spotify/confidence/ConfidenceFeatureProvider.kt [hunk 1]
```diff
@@ -258,6 +258,10 @@ class ConfidenceFeatureProvider private constructor(
         ): ConfidenceFeatureProvider {
             val configuredClient = client ?: ConfidenceRemoteClient(
                 clientSecret = clientSecret,
+                sdkId = "SDK_ID_KOTLIN_PROVIDER",
```
Comment (fabriziodemaria) [inline Provider/src/main/java/com/spotify/confidence/ConfidenceFeatureProvider.kt:261]: Discussions are happening to change the OF SDK naming from `kotlin` to `android`. Shall we follow the same route, we might want to rename this

File: release-please-config.json [hunk 2]
```diff
@@ -10,6 +10,7 @@
             "extra-files": [
                 "README.md",
                 "Provider/build.gradle.kts"
+                "Provider/src/main/java/com/spotify/confidence.ConfidenceFeatureProvider.kts"
```
Comment (fabriziodemaria) [inline release-please-config.json:13]: Would this even work? And how bad practice is this?
It's nice to imagine a solution based off of release-please that is easy and works the same way in all providers, but this method would have a few important back-draws

File: Provider/src/main/java/com/spotify/confidence/ConfidenceFeatureProvider.kt [hunk 3]
```diff
@@ -258,6 +258,10 @@ class ConfidenceFeatureProvider private constructor(
...
+                // x-release-please-start-version
+                sdkVersion = "0.1.2",
+                // x-release-please-end
```
Comment (nicklasl) [inline Provider/src/main/java/com/spotify/confidence/ConfidenceFeatureProvider.kt:264]: This was an interesting approach but I would suggest we use `BuildConfig` instead. 
You can see an example how it is defined [here](https://github.com/spotify/confidence-openfeature-provider-kotlin/blob/main/ConfidenceDemoApp/build.gradle.kts#L41) and used [here](https://github.com/spotify/confidence-openfeature-provider-kotlin/blob/8a1ece8ba4aff7a189f4d10cc38fd279d1d67b27/ConfidenceDemoApp/src/main/java/com/example/confidencedemoapp/ClientSecretProvider.kt#L4).

File: Provider/src/main/java/com/spotify/confidence/client/ConfidenceRemoteClient.kt [hunk 4]
```diff
@@ -20,6 +20,8 @@ import java.net.HttpURLConnection
 
 class ConfidenceRemoteClient : ConfidenceClient {
     private val clientSecret: String
+    private val sdkId: String
```
Comment (vahidlazio) [inline Provider/src/main/java/com/spotify/confidence/client/ConfidenceRemoteClient.kt:23]: if we add these 2 to another object (creating a new type), then we can add more stuff to it in the future if needed easier.

File: Provider/src/main/java/com/spotify/confidence/ConfidenceFeatureProvider.kt (hunk 3 above)
Comment (Billlynch) [reply Provider/src/main/java/com/spotify/confidence/ConfidenceFeatureProvider.kt:264]: you can also use the inline comment `// x-release-please-version` too

File: Provider/src/main/java/com/spotify/confidence/ConfidenceFeatureProvider.kt (hunk 1 above)
Comment (fabriziodemaria) [reply Provider/src/main/java/com/spotify/confidence/ConfidenceFeatureProvider.kt]: A problem for another PR

File: Provider/src/main/java/com/spotify/confidence/ConfidenceFeatureProvider.kt [hunk 5]
```diff
@@ -258,6 +261,7 @@ class ConfidenceFeatureProvider private constructor(
         ): ConfidenceFeatureProvider {
             val configuredClient = client ?: ConfidenceRemoteClient(
                 clientSecret = clientSecret,
+                sdkMetadata = SdkMetadata(SDK_ID, BuildConfig.SDK_VERSION),
```
Comment (fabriziodemaria) [inline Provider/src/main/java/com/spotify/confidence/ConfidenceFeatureProvider.kt:264]: Doesn't work, the result string for version is: `Build_gradle$Versions@1e4d110c.providerVersion`. To be fixed

**Review by nicklasl — APPROVED** [review]

**Pull request merged**